### PR TITLE
refactor(getPools): use multicall3

### PR DESCRIPTION
**Pull request by fadeev**
## Summary by CodeRabbit

- **New Features**
	- Enhanced pool fetching functionality to improve type safety and data clarity.
	- Integrated multicall capability for efficient batch processing of contract calls, optimizing performance.
	- Added a new JSON definition for the `aggregate` function to streamline contract interactions.

- **Bug Fixes**
	- Improved error handling for contract call data, ensuring robust management of unexpected scenarios.

### CODE REVIEW

**Comment by fadeev** [comment]
@andresaiello please, review.

**Comment by fadeev** [comment]
@zeta-chain/fullstack please, review.

File: packages/client/src/getPools.ts [hunk 1]
```diff
@@ -51,38 +58,112 @@ export const getPools = async function (this: ZetaChainClient) {
...
+    try {
+      token0 = uniswapInterface.decodeFunctionResult("token0", token0Data)[0];
+    } catch {
+      continue;
+    }
+    try {
+      token1 = uniswapInterface.decodeFunctionResult("token1", token1Data)[0];
+    } catch {
+      continue;
+    }
```
Comment (GMaiolo) [inline packages/client/src/getPools.ts:151]: We can try/catch in the same block if the granular error is not needed
```suggestion
    try {
      token0 = uniswapInterface.decodeFunctionResult("token0", token0Data)[0];
      token1 = uniswapInterface.decodeFunctionResult("token1", token1Data)[0];
    } catch {
      continue;
    }
```

Comment (GMaiolo) [inline packages/client/src/getPools.ts:151]: We can try/catch in the same block if we don't need the granular error
Maybe we can throw in the reservers in the block as well

```suggestion
    try {
      token0 = uniswapInterface.decodeFunctionResult("token0", token0Data)[0];
      token1 = uniswapInterface.decodeFunctionResult("token1", token1Data)[0];
    } catch {
      continue;
    }
```

**Review by GMaiolo — APPROVED** [review]
utACK
Left a small comment

File: packages/client/src/getPools.ts (hunk 1 above)
Comment (fadeev) [reply packages/client/src/getPools.ts]: Good _catch_ 😄 e975bef

**Pull request merged**